multicluster/service-mirror/cluster_watcher.go

```diff
@@ -148,7 +148,7 @@ func NewRemoteClusterServiceWatcher(
 			"cluster":    clusterName,
 			"apiAddress": cfg.Host,
 		}),
-		eventsQueue:  workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		eventsQueue:  workqueue.NewRateLimitingQueue(workqueue.DefaultItemBasedRateLimiter()),
```

Comment:
The `DefaultItemBasedRateLimiter` gives us a `ItemExponentialFailureRateLimiter` ( which does a simple baseDelay*2^<num-failures> limit dealing with max failures and expiration are up to the caller) with defaults as
- baseDelay 1 ms
- maxDelay 1000 s

These delays are happened during requeues for rate-limiting. Do we actually need this delay functionality as totally restrict the requeues to `requeueLimit`?